test/smem_lockstep_parity_test.cpp

```
@@ -62,14 +62,14 @@ static bool run_case(FMI_search *fmi,
 
-    int16_t *qpa_s = (int16_t *)malloc(numReads * sizeof(int16_t));
-    int16_t *qpa_l = (int16_t *)malloc(numReads * sizeof(int16_t));
+    int32_t *qpa_s = (int32_t *)malloc(numReads * sizeof(int32_t));
+    int32_t *qpa_l = (int32_t *)malloc(numReads * sizeof(int32_t));
     int32_t *mia_s = (int32_t *)malloc(numReads * sizeof(int32_t));
     int32_t *mia_l = (int32_t *)malloc(numReads * sizeof(int32_t));
     int32_t *rid_s = (int32_t *)malloc(numReads * sizeof(int32_t));
     int32_t *rid_l = (int32_t *)malloc(numReads * sizeof(int32_t));
-    memcpy(qpa_s, query_pos_array_in, numReads * sizeof(int16_t));
-    memcpy(qpa_l, query_pos_array_in, numReads * sizeof(int16_t));
+    memcpy(qpa_s, query_pos_array_in, numReads * sizeof(int32_t));
```

Cppcheck (2.21.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)